rascaline-c-api/src/lib.rs

```diff
@@ -8,7 +8,56 @@ mod utils;
 #[macro_use]
 mod status;
 pub use self::status::{catch_unwind, rascal_status_t};
+pub use log::{error, info, warn, Record, Level, Metadata, LevelFilter};
+use std::ffi::{CString};
+
 
 mod system;
 mod descriptor;
 mod calculator;
+
+pub type RascalLoggingCallback = Option<unsafe extern fn(level: i32, message: *const std::os::raw::c_char)>;
+
+
+static mut GLOBAL_CALLBACK: RascalLoggingCallback = None;
+struct RascalLogger;
+
+#[no_mangle]
+pub unsafe extern fn rascal_set_logging_callback(callback: RascalLoggingCallback) {
+    GLOBAL_CALLBACK = callback;
+    log::set_boxed_logger(Box::new(RascalLogger)).expect("Setting rascal logger faild.")
```

Comment:
This can not use expect because we want to be able to call `rascal_set_logging_callback` multiple times. Instead you can use something like OnceCell or just warn if the result is an error